service/controller/v14/resource/service/update.go

```diff
@@ -30,5 +51,41 @@ func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desire
 }
 
 func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
-	return nil, nil
+	currentServices, err := toServices(currentState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	desiredServices, err := toServices(desiredState)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out which services have to be updated")
+
+	servicesToUpdate := make([]*apiv1.Service, 0)
+
+	for _, currentService := range currentServices {
+		desiredService, err := getServiceByName(desiredServices, currentService.Name)
+		if IsNotFound(err) {
+			// Ignore here. These are handled by newDeleteChangeForUpdatePatch().
+			continue
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		if isServiceModified(desiredService, currentService) {
+			// Make a copy and set the resource version so the service can be updated.
+			serviceToUpdate := desiredService.DeepCopy()
+			serviceToUpdate.ObjectMeta.ResourceVersion = currentService.ObjectMeta.ResourceVersion
+			serviceToUpdate.Spec.ClusterIP = currentService.Spec.ClusterIP
```

Comment:
I am hitting issues in KVM when updating from `2.10.0` to `2.11.0`. The update to k8s fails and complains the service is outdated. Did you test the update from one version to another? Why do we set the resource version here explicitly?

Reply:
Ok this is bullshit. I probably got confused with the statusresource magic I hustled with meanwhile. All good. Never mind. <3